src/gui/qgsabstractrelationeditorwidget.cpp

```diff
@@ -559,7 +559,7 @@ void QgsAbstractRelationEditorWidget::unlinkFeatures( const QgsFeatureIds &fids
     for ( QgsFeatureId fid : constFeatureids )
     {
       QgsVectorLayer *referencingLayer = mRelation.referencingLayer();
-      if ( mRelation.type() == QgsRelation::Normal )
+      if ( mRelation.type() == QgsRelation::Generated )
       {
```

Comment:
should there be an else at the end of the if statement?

Reply:
I would say not, only in case the relation is generated the referenced layer field must be reset.

Reply:
oh, sorry I misread the code.